dask/array/utils.py

```diff
@@ -71,12 +75,50 @@ def meta_from_array(x, ndim=None, dtype=None):
     except Exception:
         meta = np.empty((0,) * ndim, dtype=dtype or x.dtype)
 
+    if np.isscalar(meta):
+        meta = np.array(meta)
+
     if dtype and meta.dtype != dtype:
         meta = meta.astype(dtype)
 
     return meta
 
 
+def compute_meta(func, _dtype, *args, **kwargs):
+    args_meta = list(map(meta_from_array, args))
+    kwargs_meta = {k: meta_from_array(v) if is_arraylike(v) else v for k, v in kwargs.items()}
+
+    # todo: look for alternative to this, causes issues when using map_blocks()
+    # with np.vectorize, such as dask.array.routines._isnonzero_vec().
+    if isinstance(func, np.vectorize):
+        meta = func(*args_meta)
+        if _dtype and getattr(meta, 'dtype', None) != _dtype:
+            with ignoring(AttributeError):
+                meta = meta.astype(_dtype)
+
+    else:
+        try:
+            meta = func(*args_meta, **kwargs_meta)
+        except TypeError as e:
+            if ("unexpected keyword argument" in str(e) or
+                    "is an invalid keyword for" in str(e) or
+                    "Did not understand the following kwargs" in str(e)):
+                raise
+            else:
+                return None
+        except Exception:
+            return None
+
+    if _dtype and getattr(meta, 'dtype', None) != _dtype:
+        with ignoring(AttributeError):
+            meta = meta.astype(_dtype)
+
+    if np.isscalar(meta):
+        meta = np.array(meta)
```

Comment:
I imagine this may cause problems for libraries such as CuPy, just like the entry in `meta_from_array`, but there's no test at the moment to confirm my suspicion.

Reply:
I think I've resolved this elsewhere, but I'm not entirely certain.

Reply:
Ok, let's leave it for now, we will address it if we find a failing case in the future.